### PR TITLE
set language version to allow KSP with FIR compiler

### PR DESCRIPTION
**Pull request by neetopia**
`use-k2` was deprecated.

### CODE REVIEW

**Comment by neetopia** [comment]
fixes #983

File: gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt [hunk 1]
```diff
@@ -83,6 +84,7 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
         const val KSP_GROUP_ID = "com.google.devtools.ksp"
         const val KSP_PLUGIN_CLASSPATH_CONFIGURATION_NAME = "kspPluginClasspath"
         const val KSP_PLUGIN_CLASSPATH_CONFIGURATION_NAME_NON_EMBEDDABLE = "kspPluginClasspathNonEmbeddable"
+        val LANGUAGE_VERSION = KotlinVersion.KOTLIN_1_8
```
Comment (ting-yuan) [inline gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt:87]: You can use `org.jetbrains.kotlin.config.LanguageVersion.LATEST_STABLE` until bumping to 2.0.

File: gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt [hunk 2]
```diff
@@ -415,7 +417,11 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
                         configureAsAbstractKotlinCompileTool(kspTask as AbstractKotlinCompileTool<*>)
                         configurePluginOptions(kspTask)
                         kspTask.compilerOptions.noJdk.value(kotlinCompileTask.compilerOptions.noJdk)
-                        kspTask.compilerOptions.useK2.value(false)
+                        kspTask.compilerOptions.languageVersion.orNull?.let { version ->
```
Comment (ting-yuan) [inline gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt:420]: Configuration of shared compilerOptions can be moved into a local function:
```
fun configureKotlinCompilationTask(task: KotlinCompilationTask) {
  ...
}
```

File: integration-tests/src/test/kotlin/com/google/devtools/ksp/test/PlaygroundIT.kt [hunk 3]
```diff
@@ -147,19 +146,25 @@ class PlaygroundIT {
         }
     }
 
-    // Disabled for now: ERROR: K2 does not support plugins yet, so please remove -Xuse-k2 flag
-    // Test -Xuse-fir for compilation; KSP still uses FE1.0
-    @Ignore
     @Test
     fun testFirPreview() {
-        val gradleProperties = File(project.root, "gradle.properties")
-        gradleProperties.appendText("\nkotlin.useK2=true")
```
Comment (ting-yuan) [inline integration-tests/src/test/kotlin/com/google/devtools/ksp/test/PlaygroundIT.kt:156]: Can you confirm what would happen if a user specifies `-Pkotlin.useK2=true` w. and w.o. `languageVersion`? So as to make sure removing `kspTask.compilerOptions.useK2.value(false)` in the above is safe. No need to put it into a test case.

Comment (neetopia) [reply integration-tests/src/test/kotlin/com/google/devtools/ksp/test/PlaygroundIT.kt:156]: This option is deprecated and therefore the errors we saw, it should not be set to true anymore if users want to use FIR.

Comment (ting-yuan) [reply integration-tests/src/test/kotlin/com/google/devtools/ksp/test/PlaygroundIT.kt:156]: I'd rather spend a few minutes beforehand than hours triage and blaming users' faults later.

Comment (neetopia) [reply integration-tests/src/test/kotlin/com/google/devtools/ksp/test/PlaygroundIT.kt:156]: I meant the behavior when user tries to use this flag in the setup you mentioned in a normal compilation, which is not KSP related. For KSP it doesn't matter, KSP won't get invoked in both scenarios.

File: gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt (hunk 1 above)
Comment (neetopia) [reply gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt:87]: LATEST_STABLE is 1.9 in main branch.

File: gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt (hunk 2 above)
Comment (neetopia) [reply gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt:420]: done

**Review by ting-yuan — APPROVED** [review]

**Pull request merged**